backend/inspirehep/pidstore/converters.py

```diff
@@ -4,13 +4,38 @@
 #
 # inspirehep is free software; you can redistribute it and/or modify it under
 # the terms of the MIT License; see LICENSE file for more details.
+
+from functools import partial
+
 import structlog
-from invenio_records_rest.utils import PIDConverter
+from invenio_records.api import Record
+from invenio_records_rest.utils import PIDConverter, obj_or_import_string
 from werkzeug.routing import PathConverter
 
+from .resolvers import InspireResolver
+
 LOGGER = structlog.getLogger()
 
 
+class InspirePIDConverter(PIDConverter):
+    """Just to use our own Resolver class"""
+
+    def __init__(
+        self, url_map, pid_type, getter=None, record_class=None, object_type="rec"
+    ):
+        super(PIDConverter, self).__init__(url_map)
+        getter = obj_or_import_string(
+            getter,
+            default=partial(
+                obj_or_import_string(record_class, default=Record).get_record,
+                with_deleted=True,
+            ),
+        )
+        self.resolver = InspireResolver(
+            pid_type=pid_type, object_type=object_type, getter=getter
+        )
```

Comment:
whey not something like?
```python
       super().__init__(url_map, pid_type, getter, record_class, object_type)
        self.resolver = InspireResolver(
            pid_type=pid_type, object_type=object_type, getter=getter
        )
```

Reply:
because of the line  above: `getter = obj_or_import_string(...`